deploy-oke-using-ubuntu/terraform/README.md

```diff
@@ -0,0 +1,89 @@
+
+Next, update the contents of the `terraform.tfvars` file with your values, most of which can be found in your `~/.oci/config` file. It's required to provide SSH keys as this will allow you to access the nodes for debugging purposes. Additionally, the `image_id` will be the OCID generated from registering your Ubuntu OKE image from the [Deploy OKE nodes using Ubuntu images](https://canonical-oracle.readthedocs-hosted.com/oracle-how-to/deploy-oke-nodes-using-ubuntu-images/#register-an-ubuntu-image) How-To. 
+
+The following is an example of the contents of `terraform.tfvars`:
+```
+# Required
+tenancy_ocid         = "ocid1.tenancy.oc1..xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+user_ocid            = "ocid1.user.oc1..xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+compartment_ocid     = "ocid1.compartment.oc1..xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+fingerprint          = "xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx"
+private_key_path     = "~/.oci/oci.pem"
+region               = "us-phoenix-1"
+kubernetes_version   = "v1.32.1"
+image_id = "ocid1.image.oc1.phx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+ssh_public_key_path  = "~/.ssh/id_rsa.pub"
+ssh_private_key_path = "~/.ssh/id_rsa"
+
+# Optional
+public_nodes           = false
+add_managed_nodes      = false
+add_self_managed_nodes = false
+```
+
+Before continuing on launching the cluster and creating the nodes, be sure to follow [Creating a Dynamic Group and a Policy for Self-Managed Nodes](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengdynamicgrouppolicyforselfmanagednodes.htm#contengprereqsforselfmanagednodes-accessreqs) in the [Prerequisites](#Prerequisites) section if you wish to create Self-Managed nodes. This step is required to allow the standalone VMs to have permission to join the cluster.
+
+Otherwise, you can initialize the Terraform project with:
+```bash
+terraform init
+```
+
+By default zero nodes will be created, so you must toggle which nodes you'd like to add. This can be done via `terraform.tfvars` file like so:
+```
+# Optional
+public_nodes           = false
+add_managed_nodes      = false
+add_self_managed_nodes = false
+```
+
+Alternatively, the `.tfvars` can be overridden at launch time with the following `-var` flags:
+```bash
+terraform apply -var="add_managed_nodes=true" -var="add_self_managed_nodes=true"
+```
+
+It should be noted that you are able to toggle *both* node types for the cluster at once.
+
+Once Terraform has reached it's desired state, the `kubeconfig` can be retrieved with the following:
+> Warning! This will override your current `~/.kube/config` if one already exists.
+```bash
+terraform output -json cluster_kubeconfig | yq -p json | tee ~/.kube/config
+```
+
+Finally you can verify the state of the cluster with:
+```bash
+kubectl get nodes -o wide
```

Comment:
This `kubectl` command will fail on a machine without `oci` installed since the kubeconfig we extracted above will use the `oci` tool to generate an authentication token. It might be useful to add `oci` as a dependency to the top of the page.

Reply:
You are correct, it likely tries to subprocess this and fails since it's not in PATH.
```
users:
  - name: user-xxxxxxxxxxx
    user:
      exec:
        apiVersion: client.authentication.k8s.io/v1beta1
        args:
          - ce
          - cluster
          - generate-token
          - --cluster-id
          - ocid1.cluster.oc1.xxxxxxxxxxxxxxxx
          - --region
          - <some-region>
        command: oci
        env: []

```
Added `oci` CLI as a req because of this :saluting_face: